tapiro/core/src/main/scala/io/buildo/tapiro/Http4sMeta.scala

```diff
@@ -44,17 +45,26 @@ object Http4sMeta {
     q"Router($route -> NonEmptyList($first, List(..$rest)).reduceK)"
   }
 
-  val endpoints = (routes: List[Route]) =>
-    routes.flatMap { route =>
-      val name = Term.Name(route.name.last)
-      val endpointsName = Term.Select(Term.Name("endpoints"), name)
-      val controllersName = Term.Select(Term.Name("controller"), name)
+  val endpoints = (routes: List[TapiroRoute]) =>
+    routes.map { route =>
+      val name = Term.Name(route.route.name.last)
+      val endpointsName = q"endpoints.$name"
+      val controllersName = q"controller.$name"
       val controllerContent =
-        if (route.params.length <= 1) Some(controllersName)
-        else Some(Term.Select(Term.Eta(controllersName), Term.Name("tupled")))
-      controllerContent.map { content =>
-        val toRoutes = Term.Apply(Term.Select(endpointsName, Term.Name("toRoutes")), List(content))
-        q"val ${Pat.Var(name)} = $toRoutes"
-      }
+        route.method match {
+          case RouteMethod.GET =>
+            route.route.params.length match {
+              case 0 => q"_ => $controllersName()"
+              case 1 => controllersName
+              case _ => q"($controllersName _).tupled"
+            }
+          case RouteMethod.POST =>
+            val fields = route.route.params
+              .filterNot(_.tpe == MetarpheusType.Name("AuthToken"))
+              .map(p => Term.Name(p.name.getOrElse(Meta.typeNameString(p.tpe))))
+            q"x => $controllersName(..${fields.map(f => q"x.$f")})"
+        }
+      val toRoutes = q"$endpointsName.toRoutes($controllerContent)"
+      q"val ${Pat.Var(name)} = $toRoutes"
```

Comment:
this seems identical to the same code in `AkkaHttpMeta`. Can we deduplicate it? (Not necessarily in this PR)

Reply:
I've factored out most of the duplication while fixing bugs

There's still some duplicated code here and there that processes route parameters (filtering out the auth param, computing default param names from their types if missing, etc.) which I think we should factor out, but it's best done in another PR I think

In particular, I think we could include in `TapiroRoute` a "cleaned" version of the route parameters, already processed so we don't have to do the same processing in different functions